File: Backend/Organizer/Models/Registration.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Organizer.Models
{
    [Table("registration")]
    public class Registration
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column(name:"reg_id")]
        public int RegId { get; set; }  // PK, AI

        [Required]
        public int Eid { get; set; }    // FK → event.eid

        [Required]
        [Column(name: "att_id")]
        public int AttId { get; set; }  // FK → attendee.att_id

        [Required]
        [Range(1, int.MaxValue)]
        [Column(name: "no_of_people")]
        public int NoOfPeople { get; set; } // NOT NULL
    }

}

File: Backend/Attendee/Attendee/src/main/java/com/example/attendee/entity/Registration.java
package com.example.attendee.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "registration")
@Getter
@Setter
public class Registration {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "reg_id")
    private int regId;

    // Foreign key to event.eid
    private int eid;

    // Foreign key to attendee.att_id
    @Column(name = "att_id")
    private int attId;

    @Column(name = "no_of_people")
    private int noOfPeople;
}

